arronizMaria/Building.java

```diff
@@ -0,0 +1,127 @@
+                ==============================
+                ==================================
+
+                """;
+        final String TIP = "     ".repeat(3) + "__/\\__";
+        final String HEADER = "|####|";
+        final String ANTENNA = "  |  ";
+        final String ROOF = "=".repeat(36);
+
+        for (int day = 1; day < END_OF_WEEK; day++) {
+            for (int hour = 0; hour < HOURS_PER_DAY; hour++) {
+
+                dailyConsume = dailyConsume + consumePerHour;
+                consumePerHour = 0;
+
+                boolean lightingStrikes = false;
+                boolean doingMaintenance = false;
+                double lighting = Math.random();
+                double maintenance = Math.random();
+
+                if (!lightingStrikes && lighting <= PROB_LIGHTING) {
+                    column = (int) (Math.random() * 6);
+                    lightingStrikes = true;
+                }
+                if (!doingMaintenance && maintenance <= PROB_MAINTENANCE) {
+                    row = (int) (Math.random() * 6);
+                    doingMaintenance = true;
+                }
+
+                cleanScreen();
+                System.out.println(TIP);
+                System.out.println(ANTENNA.repeat(3) + HEADER + ANTENNA.repeat(3));
+                System.out.println(ROOF);
+
+                for (int floor = 0; floor < MAX_FLOORS; floor++) {
+                    for (int window = 0; window < MAX_WINDOWS; window++) {
+
+                        double windowOpen = Math.random();
+                        double lightsOn = Math.random();
+
+                        if (windowOpen <= PROB_WINDOW_OPENED) {
+                            isOpen = true;
+                        } else {
+                            isOpen = false;
+                        }
+                        if (lightsOn <= PROB_LIGHTS_ON) {
+                            isOn = true;
+                            consumePerHour++;
+                        } else {
+                            isOn = false;
+                        }
+
+                        String theWindow = printWindow(isOn, isOpen);
+
+                        if (window == column) {
+                            theWindow = ":[X]:";
+                        }
+                        if (floor == row) {
+                            theWindow = ":[#]:";
+                        }
+
+                        System.out.print(theWindow);
+
+                    }
+                    System.out.println();
+                }
+                System.out.print(basement);
+
+                printStatistics(day, hour, consumePerHour, column, row, dailyConsume);
+
+                pause(1);
+            }
+        }
+    }
+
+    private static String printWindow(boolean isOn, boolean isOpen) {
+        if (!isOpen) {
+            return ":[ ]:";
+        } else if (isOn) {
+            return ":[*]:";
+        }
+        return ":[']:";
+
+    }
```

Comment:
```suggestion
    private static String printWindow(boolean isOn, boolean isOpen) {
        if (isOpen) {
            if (isOn){
                return VENTANA_ENCENDIDA;
            } else {
                return VENTANA_APAGADA;
            }
        }
        return VENTANA_CERRADA;
    }
```

A la última línea solo llega si está cerrada. Y podría quedar aún mas corto con una combinación de if y ternario:

```suggestion
    private static String printWindow(boolean isOn, boolean isOpen) {
        if (isOpen) {
                return isOn ? VENTANA_ENCENDIDA : VENTANA_APAGADA;
        }
        return VENTANA_CERRADA;
    }
```